src/types/routePaths.ts

```diff
@@ -0,0 +1,55 @@
+import { ExtractRouteMethodParams } from '@/types/routeMethods'
+import { Public, Route, Routes } from '@/types/routes'
+
+export type ExtractRoutePathParameters<
+  TRouteMethods,
+  TRoutePath extends string
+> = TRoutePath extends string ? NestedObjectValue<TRouteMethods, TRoutePath> : never
+
+type NestedObjectValue<
+  TRouteMethods,
+  TRoutePath extends string
+> =
+  TRoutePath extends `${infer F}.${infer R}`
+    ? F extends keyof TRouteMethods
+      ? NestedObjectValue<TRouteMethods[F], R>
+      : never
+    : TRoutePath extends keyof TRouteMethods
+      ? ExtractRouteMethodParams<TRouteMethods[TRoutePath]>
+      : never
```

Comment:
bit of a nit but we can avoid one of these nested ternaries by checking extends after the inference of F

```suggestion
  TRoutePath extends `${infer F extends string & keyof TRouteMethods}.${infer R}`
    ? NestedObjectValue<TRouteMethods[F], R>
    : TRoutePath extends keyof TRouteMethods
      ? ExtractRouteMethodParams<TRouteMethods[TRoutePath]>
      : never
```